src/viam/sdk/common/exception.hpp

```diff
@@ -0,0 +1,66 @@
+/// @file common/exception.hpp
+///
+/// @brief Defines custom exceptions for the SDK.
+#pragma once
+#include <grpcpp/support/status.h>
```

Comment:
I want to think about how I feel about having Status become an embedded part of our API.

Reply:
Another option would be to have the error details, code and message as fields instead of a `grpc::Status`.